src/ProjectTemplates/BlazorTemplates.Tests/PlaywrightFixture.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging.Testing;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace ProjectTemplates.Tests.Infrastructure
+{
+    public class PlaywrightFixture<TTestAssemblyType> : IAsyncLifetime
+    {
+        private static readonly bool _isCIEnvironment =
+            !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("ContinuousIntegrationBuild"));
+
+        private readonly IMessageSink _diagnosticsMessageSink;
+        private readonly IConfiguration _configuration;
+
+        public PlaywrightFixture(IMessageSink diagnosticsMessageSink)
+        {
+            _diagnosticsMessageSink = diagnosticsMessageSink;
+
+            _configuration = CreateConfiguration(typeof(TTestAssemblyType).Assembly);
+        }
+
+        private IConfiguration CreateConfiguration(Assembly assembly)
+        {
+            var basePath = Path.GetDirectoryName(assembly.Location);
+            var os = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "win" :
+                RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "linux" :
+                RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? "osx" : null;
```

Comment:
You could use this: https://docs.microsoft.com/en-us/dotnet/api/system.platformid?view=net-5.0

Reply:
Done